app/api/helpers/user.py

```diff
@@ -0,0 +1,49 @@
+from sqlalchemy.orm.exc import NoResultFound
+from app.models import db
+from app.api.helpers.db import save_to_db
+from app.models.user import User
+
+
+def modify_email_for_user_to_be_deleted(user):
+    """
+    Update the email ID of user which is to be deleted.
+    Adds '.deleted' substring to email of a user to be deleted.
+    :param order: User to be deleted.
+    :return:
+    """
+    notUniqueEmail = True
+    userEmail = user.email
+    while notUniqueEmail:
+        try:
+            db.session.query(User).filter_by(email=userEmail).one()
+        except NoResultFound:
+            user.email = userEmail
+            save_to_db(user)
+            notUniqueEmail = False
+        else:
+            userEmail = userEmail + '.deleted'
+    return user
+
+
+def modify_email_for_user_to_be_restored(user):
+    """
+    Update the email ID of user which is to be restored.
+    Removes '.deleted' substring from a user to be restored.
+    :param order: User to be deleted.
+    :return:
+    """
+    notUniqueEmail = True
+    userEmail = user.email
+    while notUniqueEmail:
+        try:
+            db.session.query(User).filter_by(email=userEmail).one()
+        except NoResultFound:
+            user.email = userEmail
+            save_to_db(user)
+            notUniqueEmail = False
+        else:
+            if userEmail.endswith('.deleted'):
+                userEmail = userEmail[:-8]
+            else:
+                userEmail = userEmail + '.new'
```

Comment:
Why?

Reply:
Suppose a user is restored but he/she already signed up so the constraint is violated as 2 users cannot have same email. So I just added a '.new' to the email just for this sake.
This case however may happen only once or twice

Reply:
This should not be automated. Admin should first change the email manually to something and then restore

Reply:
@iamareebjamal I'll remove this condition and if you want I can open a separate issue to handle this. 
>when I update the email of user in modify_email_for_user_to_be_restored function, it is not getting reflected on database

In the meantime can you please give this a look? ^^

Reply:
@iamareebjamal any help?

Reply:
Will look tomorrow. @prateekj117 @uds5501 Can you please take a look as well?

Reply:
@iamareebjamal @shreyanshdwivedi will take a look tonight

Reply:
@shreyanshdwivedi The code looks correct. Any error logs during implementation?

Reply:
No error logs. @uds5501
@iamareebjamal can you take a look as well please